Code style: Sniff operator spacing

elkuku opened the pull request:
This ensures that operators are surrounded by one space.

elkuku committed:
Code style: Add OperatorSpacingSniff

elkuku committed:
Code style: Fix operator spacing

joomla-jenkins commented:
Unit testing complete.  There were 0 failures and 0 errors from 1872 tests and 11090 assertions.
Checkstyle analysis reported 235 warnings and 0 errors.

eddieajau commented on libraries/joomla/application/component/controlleradmin.php:
```diff
@@ -225,7 +225,7 @@ class JControllerAdmin extends JController
 		// Initialise variables.
 		$ids = JRequest::getVar('cid', null, 'post', 'array');
-		$inc = ($this->getTask() == 'orderup') ? -1 : +1;
+		$inc = ($this->getTask() == 'orderup') ? - 1 : + 1;
```
Maybe it's just me, but I don't like the space where it applies to a negative number.

elkuku committed:
Code style: Fix OperatorSpacingSniff

The Sniff was missing the check for a ternary "?" and also wasn't
prepared for positives like +1

elkuku committed:
Code style: Revert false positives

elkuku commented:
Actually I wasn't quite sure myself when I changed it.
There was a man who drove his car into a lake because his navigator told him to go straight. Don't trust the machines ;)

So I changed the machine, err the sniff, to look if there is a ternary "?" before the operator, and also that people might use a + sign for a positive number ;)

eddieajau commented:
Just a side note, I'm fixing up all the missing "public" stuff in the functions.

elkuku commented:
I think this is related to one of the other pulls ;)
But yes, good stuff. When it is done I believe the sniff should give an error instead of a warning on missing scope modifiers.

eddieajau commented:
Sounds good.

joomla-jenkins commented:
Build triggered by changes to the head.

Unit testing complete.  There were 0 failures and 0 errors from 1873 tests and 11091 assertions.
Checkstyle analysis reported 552 warnings and 1 errors.

elkuku committed:
Merge remote branch 'origin/master' into sniff-operator-spacing

LouisLandry commented:
Is this going to allow for people to have assignment operations line up?

``` php
$foo    = 'bar';
$foobar = 'snafu';
$bar    = 'high';
```

?

elkuku commented:
Yes, this is still allowed.

LouisLandry commented:
Great, thanks.  Is it worth going over things and also fixing all the new issues here?  There appears to be at least one error.

elkuku committed:
Merge remote branch 'origin/master' into sniff-operator-spacing

elkuku committed:
Code style: Fix operator spacing

elkuku commented:
Well, I would be curious about honorable Mr. Jenkins error.. I ran my "personal" jenkins, without errors. Could be a hickup :P
http://elkuku.github.com/pulltester/pulls/572

> all the new issues ?

I merged the master, found 1 (one) new error, fixed it, pushed, let's see what Mr. Jenkins says ;)

joomla-jenkins commented:
Build triggered by changes to the head.

Unit testing complete.  There were 0 failures and 0 errors from 1971 tests and 11142 assertions.
Checkstyle analysis reported 199 warnings and 0 errors.
